File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task145_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.Scanner;

public class Main {
    public static int maxSubarraySum(int[] arr) {
        int maxSum = Integer.MIN_VALUE;
        int currentSum = 0;
        for (int num : arr) {
            currentSum = Math.max(num, currentSum + num);
            maxSum = Math.max(maxSum, currentSum);
        }
        return maxSum;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the array elements separated by space: ");
        String[] input = scanner.nextLine().split(" ");
        int[] arr = new int[input.length];
        for (int i = 0; i < input.length; i++) {
            arr[i] = Integer.parseInt(input[i]);
        }
        System.out.println("Maximum contiguous sum is " + maxSubarraySum(arr));
    }
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task145_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int maxSubarraySum(std::vector<int>& arr) {
    int maxSum = INT_MIN;
    int currentSum = 0;
    for (int num : arr) {
        currentSum = std::max(num, currentSum + num);
        maxSum = std::max(maxSum, currentSum);
    }
    return maxSum;
}

int main() {
    std::vector<int> arr;
    int num;
    std::cout << "Enter the array elements separated by space: ";
    while (std::cin >> num) {
        arr.push_back(num);
        if (std::cin.peek() == '
') break;
    }
    std::cout << "Maximum contiguous sum is " << maxSubarraySum(arr) << std::endl;
    return 0;
}